pkg/svcat/service-catalog/class.go

```diff
@@ -27,16 +27,32 @@ import (
 const (
 	// FieldExternalClassName is the jsonpath to a class's external name.
 	FieldExternalClassName = "spec.externalName"
+
+	// FieldClusterServiceBrokerName is the json path to a classes associated broker
+	FieldClusterServiceBrokerName = "spec.clusterServiceBrokerName"
 )
 
 // RetrieveClasses lists all classes defined in the cluster.
-func (sdk *SDK) RetrieveClasses() ([]v1beta1.ClusterServiceClass, error) {
-	classes, err := sdk.ServiceCatalog().ClusterServiceClasses().List(v1.ListOptions{})
+func (sdk *SDK) RetrieveClasses(opts *FilterOptions) ([]v1beta1.ClusterServiceClass, error) {
+	// Not filtering by broker, just return classes
+	if opts.Broker == "" {
```

Comment:
Instead of the big if else, I recommend doing something like this:

1. define the list options
2. check if each relevant opt is != "" and add it to the field selectors
3. do a single list call that uses the list options